File: lab/python/textos/extrair_dominio_de_email.py
import re


def is_email(email):
    return bool(re.match(r"^[\w\.-]+@[\w\.-]+\.\w{2,}$", email))


def extract_dominio(email):
    if not is_email(email):
        return "Não é EMAIL"
    match = re.search(r"@([\w.-]+\.[a-zA-Z]{2,})$", email)
    return match.group(1) if match else "Domínio não encontrado"


data_emails = [
    'teste@email.com.br',
    "olamundo.com",
    "jvj12@teste.com",
    "teste@gmail.com"
]

for data_email in data_emails:
    print(f"{data_email:25} -> {extract_dominio(data_email)}")

File: lab/javascript/textos/extrair_dominio_de_email.js
function is_email(email) {
  const regex = /^[\w.-]+@[\w.-]+\.\w{2,}$/;
  return regex.test(email);
}

function extract_dominio(email) {
  if (!is_email(email)) return "Não é EMAIL";

  const match = email.match(/@([\w.-]+\.[a-zA-Z]{2,})$/);
  return match ? match[1] : "Domínio não encontrado";
}

// -------------------- Testes --------------------
const data_emails = [
  "teste@email.com.br",
  "olamundo.com",
  "jvj12@teste.com",
  "teste@gmail.com"
];

for (const data_email of data_emails) {
  console.log(`${data_email.padEnd(25)} -> ${extract_dominio(data_email)}`);
}
